File: Flipping bits/Flipping bits.py
def flippingBits(n):
    #print(n)
    binary = bin(n)
    binary = binary[2:] 
    #print( len(binary) )
    
    if len(binary) < 32:
        binary = ((32-len(binary))*'0')+binary
        #print(binary)
        
    binary2 =""
    for index, x in enumerate(binary):
        if x == '0':
            binary2+='1'
        else:
            binary2+='0'
    
    #print(binary2)                
    #print(int(binary2,2))
    
    return (int(binary2,2))


##  Done

File: Flipping bits/Flipping_bits.cpp
#include<iostream>
#include<vector>
#include<string>
#include <bitset>

using namespace std;

long flippingBits(long n) {
    bitset<32> dec_to_binary(n); 
    //cout<<"binary converstion - "<<dec_to_binary<<endl;
    for(int i = 0; i < dec_to_binary.size(); i++)
    {
        dec_to_binary[i] = !dec_to_binary[i];
    }
    //cout<<"flipped binary will be - "<<dec_to_binary<<endl;
    //cout<<"decimal converstion - "<<dec_to_binary.to_ulong();
    return dec_to_binary.to_ulong();
}

int main(){

	long num = 4;
	cout<<flippingBits();

	return 0;
}

//done